src/functions/uploadToRSIS/framework/repo/rsis-repository.ts

```diff
@@ -40,3 +45,8 @@ export const saveTestResult = async (
     info(`This is where we would be issuing the following SQL statement:\n${insertSql}\nUsing params:\n${params}`);
   }
 };
+
+export const getTableNameByTestCategory = (testCategory: CategoryCode): string => {
+  const category = testCategory as TestCategory;
+  return ([TestCategory.CCPC, TestCategory.DCPC].indexOf(category) >= 0 ? 'cpc4_holding' : 'dl25mes_holding');
+};
```

Comment:
export const getTableNameByTestCategory = (testCategory: CategoryCode): string => {
  const category = testCategory as TestCategory;
  return ([TestCategory.CCPC, TestCategory.DCPC].indexOf(category) >= 0 ? 'cpc4_holding' : 'dl25mes_holding');
};

possibly simplify to:

export const getTableNameByTestCategory = (categoryCode: CategoryCode): string => {
  return [TestCategory.CCPC, TestCategory.DCPC].indexOf(categoryCode) >= 0 ? 'cpc4_holding' : 'dl25mes_holding';
};

Reply:
I get type errors unless I cast as TestCategory

Reply:
updated